src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,124 @@
+        try {
+            if (input.startsWith("todo") || input.startsWith("deadline") || input.startsWith("event")) {
+                if (input.startsWith("todo")) {
+                    if (!input.equals("todo")) {
+                        String newInput = input.replace("todo ", "");
+                        if (newInput.trim().length() == 0) {
+                            throw new DukeToDoIncorrectException();
+                        } else {
+                            TodoTask todoTask = new TodoTask(newInput);
+                            taskList.add(todoTask);
+                            newTask = todoTask;
+                        }
+                    } else {
+                        throw new DukeToDoIncorrectException();
+                    }
+                } else {
+                    try {
+                        String description = input.substring(0, input.toString().indexOf("/") - 1);
+                        String date = input.substring(input.lastIndexOf("/") + 4);
+
+                        if (input.startsWith("deadline")) {
+                            String newDescription = description.replace("deadline ", "");
+                            DeadlineTask deadlineTask = new DeadlineTask(newDescription, date);
+                            taskList.add(deadlineTask);
+                            newTask = deadlineTask;
+                        } else if (input.startsWith("event")) {
+                            String newDescription = description.replace("event ", "");
+                            EventTask eventTask = new EventTask(newDescription, date);
+                            taskList.add(eventTask);
+                            newTask = eventTask;
+                        }
+                    } catch (StringIndexOutOfBoundsException e) {
+                        if (input.contains("deadline")) {
+                            throw new DukeDeadlineIncorrectException();
+                        } else {
+                            throw new DukeEventIncorrectException();
+                        }
+                    }
+                }
+            } else {
+                throw new DukeUnknownException();
+            }
+            return Ui.addedMsg(newTask);
+
+        } catch (DukeException e) {
+            return (e.getMessage());
+        }
+    }
+
+    public static String delete(String input) {
+        if (!input.equals("delete")) {
+            try {
+                int index = Integer.parseInt(input.substring(7)) - 1;
+                Task removedTask = taskList.remove(index);
+                return Ui.deleteMsg(removedTask);
+            } catch (IndexOutOfBoundsException e) {
+                return (new DukeException("ERROR: there is no item in the list there!").getMessage());
+            }
+        } else {
+            return (new DukeException("ERROR: please specify an index.").getMessage());
+        }
+    }
+
+    /**
+     * Method to show contents of list.
+     * @return
```

Comment:
Missing return description. 
```suggestion
     * @return Contents of list as a string.
```